hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Listing.java

```diff
@@ -68,15 +68,21 @@
  * Place for the S3A listing classes; keeps all the small classes under control.
  */
 @InterfaceAudience.Private
-public class Listing extends AbstractStoreOperation {
+public class Listing {
```

Comment:
1. I'd keep it as a subclass of AbstractStoreOperation.
2. we may want to declare it as Closeable, and in tests and S3AFS call close()/closeQuietly() on it. Gives us the option of doing more with it later

What I don't want -and which you have done in this commit, is add the new list commands into the OperationCallbacks interface